src/TuxAnimations.js

```diff
@@ -45,75 +46,131 @@ var makeAnimation = function (transitions, customClassName) {
+            if (duration) {
+              //Case to accept if duration prop is a number
+              if (typeof duration === "number") {
+                duration = duration + "ms";
+              }
+              transitions[action]['transition-duration'] = duration;
             }
-            transitions[action]['transition-duration'] = duration;
-          }
 
-          //Easing prop work-------------------------------------------
-          var easing = this.props.easing;
-          //Use default if no easing prop defined
-          if (easing) {
-            //Check to see if easing exists in default Easings object from the AnimationEasings module
-            if (easing in Easings) {
-              //Set easing to value of correpsonding key from Easings module object
-              easing = Easings[easing];
+            //Easing prop work-------------------------------------------
+            var easing = this.props.easing;
+            //Use default if no easing prop defined
+            if (easing) {
+              //Check to see if easing exists in default Easings object from the AnimationEasings module
+              if (easing in Easings) {
+                //Set easing to value of correpsonding key from Easings module object
+                easing = Easings[easing];
+              }
+              transitions[action]['transition-timing-function'] =  easing;
             }
-            transitions[action]['transition-timing-function'] =  easing;
-          }
 
-          //Delay prop work--------------------------------------------
-          var delay = this.props.delay;
-          //Use default if no delay prop defined
-          if (delay) {
-            //Case to accept if delay prop is a number
-            if (typeof delay === "number") {
-              delay = delay + "ms";
+            //Delay prop work--------------------------------------------
+            var delay = this.props.delay;
+            //Use default if no delay prop defined
+            if (delay) {
+              //Case to accept if delay prop is a number
+              if (typeof delay === "number") {
+                delay = delay + "ms";
+              }
+              transitions[action]['transition-delay'] =  delay;
             }
-            transitions[action]['transition-delay'] =  delay;
-          }
 
-          //Custom properties work-------------------------------------
-          var custom = this.props.custom;
-          if (custom) {
-            for (var cssElement in custom) {
-              transitions['enterActive'][cssElement] =  custom[cssElement];
-              transitions['leave'][cssElement] =  custom[cssElement];
+            //Custom prop work-------------------------------------
+            var custom = this.props.custom;
+            if (custom) {
+              for (var cssElement in custom) {
+                transitions['enter-active'][cssElement] =  custom[cssElement];
```

Comment:
Store `transitions['enter-active'] and `transitions['leave']` in a variable before looping so you don't have to look it up each time.  Store `custom[cssElement]` in a variable so you don't have to look it up twice.